src/base/moduledownloader.lua

```diff
@@ -0,0 +1,125 @@
+--
+-- moduledownloader.lua
+-- Downloads a module from a package server
+-- Copyright (c) 2002-2016 Jason Perkins and the Premake project
+--
+
+
+---
+-- progress bar.
+--
+	local function _http_progress(total, current)
+		local width = 78
+		local progress = math.floor(current * width / total)
+
+		if progress == width then
+			io.write(string.rep(' ', width + 2) .. '\r')
+		else
+			io.write('[' .. string.rep('=', progress) .. string.rep(' ', width - progress) .. ']\r')
+		end
+	end
+
+---
+-- escape parameters in a url.
+--
+	local function escape_url_param(param)
```

Comment:
It might... I don't know ;) this has been in our package system for a long time now, I've just been copying small bits and pieces to get this module system off the ground, an idea I've had for a few months now. (to do it through the require).